### PR TITLE
Add Keras docker image

### PR DESCRIPTION
**Pull request by osanseviero**
This is an initial implementation of the Keras image. It has some strong assumptions (such as the full model being saved), but I would like to merge an initial working implementation asap and follow-up with image-captioning. 

@Rocketknight1 could you take a quick look at `pipelines/image_classification.py` please?

### CODE REVIEW

File: api-inference-community/docker_images/keras/app/pipelines/image_classification.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            inputs = inputs.convert("L")
+        target_size = (expected_input_size[1], expected_input_size[2])
+        img = inputs.resize(target_size)
+        img_array = tf.keras.preprocessing.image.img_to_array(img)
+        img_array = tf.expand_dims(img_array, 0)
+
+        predictions = self.model.predict(img_array)
+
+        if self.single_output_unit:
+            score = predictions[0][0]
```
Comment (Rocketknight1) [inline api-inference-community/docker_images/keras/app/pipelines/image_classification.py:57]: The outputs don't have softmax or sigmoid applied inside the model, right? We're not doing it in post-processing either, so that means we'll be returning un-normalized scores to the user - which is probably fine, but just making sure that's the expected behaviour!

**Review by Rocketknight1 — APPROVED** [review]
I've only looked at image_classification.py, but it looks good to me!

File: api-inference-community/docker_images/keras/app/pipelines/image_classification.py (hunk 1 above)
Comment (osanseviero) [reply api-inference-community/docker_images/keras/app/pipelines/image_classification.py:57]: The less assumptions we make, the better.

For the two models I tested, I already had softmax/sigmoid baked in the model. I think that the model should be as self contained as possible and hence we should not make assumptions. I'm not sure how this works with the widget though :)

**Comment by osanseviero** [comment]
Friendly ping @Narsil.

File: api-inference-community/docker_images/keras/app/pipelines/image_classification.py (hunk 1 above)
Comment (Narsil) [reply api-inference-community/docker_images/keras/app/pipelines/image_classification.py:57]: Widgets don't assume anything so you're good !

File: api-inference-community/docker_images/keras/app/pipelines/image_classification.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            labels = [
+                {"label": self.id2label["1"], "score": str(score)},
+                {"label": self.id2label["0"], "score": str(1 - score)},
+            ]
+        else:
+            labels = [
+                {"label": self.id2label[str(i)], "score": score.item()}
+                for i, score in enumerate(predictions[0])
+            ]
```
Comment (Narsil) [inline api-inference-community/docker_images/keras/app/pipelines/image_classification.py:66]: Nit: Maybe sorting by score might be a slight improvement. (not done within transformers, but we do regret it, could be api-inference-community handling that too tbh.

File: api-inference-community/tests/test_dockers.py [hunk 3]
```diff
@@ -112,6 +112,9 @@ def test_speechbrain(self):
     def test_timm(self):
         self.framework_docker_test("timm", "image-classification", "sgugger/resnet50d")
 
+    def test_keras(self):
+        self.framework_docker_test("timm", "image-classification", "osanseviero/")
```
Comment (Narsil) [inline api-inference-community/tests/test_dockers.py:116]: Is timm correct ?

**Review by Narsil — COMMENTED** [review]
Looks good, but we probably need to test the real docker image with a good model for that.

File: api-inference-community/docker_images/keras/app/pipelines/image_classification.py (hunk 2 above)
Comment (osanseviero) [reply api-inference-community/docker_images/keras/app/pipelines/image_classification.py:66]: Done

**Comment by osanseviero** [comment]
FYI @nateraw

File: api-inference-community/docker_images/keras/app/pipelines/image_classification.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            A :obj:`list`:. The list contains items that are dicts should be liked {"label": "XXX", "score": 0.82}
+                It is preferred if the returned list is in decreasing `score` order
+        """
+        # Resize image to expected size
+        expected_input_size = self.model.input_shape
+        if expected_input_size[-1] == 1:  # Single channel, we assume grayscale
+            inputs = inputs.convert("L")
+        target_size = (expected_input_size[1], expected_input_size[2])
+        img = inputs.resize(target_size)
+        img_array = tf.keras.preprocessing.image.img_to_array(img)
```
Comment (nateraw) [inline api-inference-community/docker_images/keras/app/pipelines/image_classification.py:50]: Nice! This is clean. We can think up a way to use a `feature_extractor` in the future, which would enable users to apply applicable transforms required by their model here. This is great for now!!

Comment (osanseviero) [reply api-inference-community/docker_images/keras/app/pipelines/image_classification.py:50]: Yeah, I think there's a lot we can experiment with here. This is mostly a proof of concept for the time being :) 

Thanks!

**Comment by Narsil** [comment]
@osanseviero had opportunity to check, everything seems good, let's merge ! :D

**Review by Narsil — APPROVED** [review]
